File: dotnet/src/DocGen/Exceptions/ServiceUnavailableException.cs
namespace Dokmatiq.DocGen.Exceptions;

/// <summary>
/// Exception thrown on HTTP 503 – service temporarily unavailable.
/// </summary>
public class ServiceUnavailableException : ApiException
{
    public ServiceUnavailableException(string message, string? responseBody)
        : base(503, message, responseBody) { }
}

File: php/src/Exception/ServiceUnavailableException.php
<?php

declare(strict_types=1);

namespace Dokmatiq\DocGen\Exception;

/** 503 – Service unavailable (e.g. LibreOffice pool exhausted). */
class ServiceUnavailableException extends ApiException
{
    public function __construct(string $message = 'Service unavailable', ?string $responseBody = null)
    {
        parent::__construct(503, $message, $responseBody);
    }
}
